articles/prototyping-machine-learning-models-with-streamlit/index.md

```diff
@@ -0,0 +1,185 @@
+from sklearn.metrics import confusion_matrix
+import numpy as np
+import itertools
+import matplotlib.pyplot as plt
+```
+After that let’s add a header for our web app and a subheader and to do that we simply use the code below
+
+```python
+st.title("Sentiment Analyzer Based On Text Analysis ")
+st.subheader("Paras Patidar - MLAIT")
+st.write('\n\n')
+```
+The next step will be to write a function that will get all the data from our directory and bring them up,
+
+```python
+@st.cache
+def get_all_data():
+    root = "Datasets/"
+    with open(root + "imdb_labelled.txt", "r") as text_file:
+        data = text_file.read().split('\n')
+         
+    with open(root + "amazon_cells_labelled.txt", "r") as text_file:
+        data += text_file.read().split('\n')
+
+    with open(root + "yelp_labelled.txt", "r") as text_file:
+        data += text_file.read().split('\n')
+
+    return data
+```
+If you notice in the function above the first line is *@st.cache*, this mark tells streamlit that whenever the function is called that there are some certain requirements it should check for and they should be met, now create a variable called all_data which should call the *get_all_data* function we just wrote.
+
+```python
+all_data = get_all_data()
+```
+Next, we create a checkbox using the *st.checkbox* function that will display Show dataset and this checkbox should have the value from the *all_data* variable we created earlier.
+
+```python
+if st.checkbox('Show Dataset'):
+    st.write(all_data)
+```
+Now we write another function that will carry out data preprocessing and to achieve that we use the code below.
+
+```python
+@st.cache
+def preprocessing_data(data):
+    processing_data = []
+    for single_data in data:
+        if len(single_data.split("\t")) == 2 and single_data.split("\t")[1] != "":
+            processing_data.append(single_data.split("\t"))
+
+    return processing_data
+```
+
+Now let’s create another checkbox that will display *show_preprocessed_data* and will get the data from the *all_data* function and pass it through the preprocessing data function we wrote above. 
+```python
+if st.checkbox('Show PreProcessed Dataset'):
+    st.write(preprocessing_data(all_data))
+```
+
+The next step will be writing a function that will split the data and call it *split_data*
+
+```python
+@st.cache
+def split_data(data):
+    total = len(data)
+    training_ratio = 0.75
+    training_data= []
+    evaluation_data = []
+
+    for indice in range(0,total):
+        if indice<total*training_ratio:
+            training_data.append(data[indice])
+        else:
+            evaluation_data.append(data[indice])
+
+    return training_data, evaluation_data
+```
+
+Now we are going to create two functions, one will be *preprocessing_step* that will get the data, preprocess it, and then split it, and the second one will be *training_step* that will receive two parameters and vectorize the training text.
```

Comment:
You also have to explain the machine learning part of it. What is happening in each function? I see you fit the vectorizer with training data but didn't understand what happened with the bernoulliNB function.